### PR TITLE
Lio support

### PR DESCRIPTION
**Pull request by Arau**

### CODE REVIEW

File: INSTALL_STORE_CONTAINER.md [hunk 1]
```diff
@@ -65,27 +65,13 @@ The container installation method requires Docker 1.10+ running on 64-bit linux.
 
 The node container (or plugin) should be installed on each Docker node where you want to consume StorageOS volumes or to present capacity to other nodes.
 
-### Network Block Device (NBD)
+### Linux-IO (LIO)
 
-(Optional) NBD is a default Linux kernel module that allows block devices to be run in userspace. Enabling NBD is recommended as it will increase performance for some workloads. To enable the module and increase the number of allowable devices, you must either run:
+StorageOS uses FUSE known as a Filesystem in Userspace. It is a software interface for Unix-like computer operating systems that lets non-privileged users create their own file systems without editing kernel code. In addition, StorageOS is developed using Linux-IO (LIO) Target. An open-source implementation of the SCSI target.
```
Comment (croomes) [inline INSTALL_STORE_CONTAINER.md]: I'd talk more about LIO here as that is what the user will see - fuse is pretty much hidden and could cause confusion.  Besides, we don't use it in the way other's are used to.

File: _docs/install/prerequisites/devicepresentation.md [hunk 2]
```diff
@@ -7,51 +7,13 @@ module: install/prerequisites/devicepresentaion.md
 
 # Device presentation
 
-StorageOS requires one of two runtime requirements to present virtual block devices
-from userspace. While not mandatory, the NBD aproach does offer significantly
-better performance over the loop-device alternative.
+StorageOS is developed using Linux-IO (LIO) Target. An open-source implementation of the SCSI target. In addition, StorageOS uses FUSE, known as a Filesystem in Userspace. It is a
```
Comment (croomes) [inline _docs/install/prerequisites/devicepresentation.md]: same, remove fuse reference.

File: _docs/reference/os_support.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
+---
+layout: guide
+title: StorageOS Docs - OS support
+anchor: reference
+module: reference/ossupport
+---
+
+# OS support
+
+StorageOS uses FUSE known as a Filesystem in Userspace. It is a software interface for Unix-like computer operating systems that lets non-privileged users create their own file systems without editing kernel code. In addition, StorageOS is developed using Linux-IO (LIO) Target. An open-source implementation of the SCSI target.
```
Comment (croomes) [inline _docs/reference/os_support.md]: same

File: _docs/reference/os_support.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+However, LIO is supported by most of the kernels available nowadays, some distributions have left the kernel module out of the main kernel package.
+
+
+## Distribution support
+
+The following table shows which distributions can enable the kernel modules to use LIO, weather needed to install linux-virtual-extra or not.
```
Comment (croomes) [inline _docs/reference/os_support.md]: weather -> whether

File: INSTALL_STORE_CONTAINER.md (hunk 1 above)
Comment (Arau) [reply INSTALL_STORE_CONTAINER.md]: fixed https://github.com/storageos/storageos.github.io/pull/148/commits/93a3652b57840070c18b0153615fe2568a9fcfb2

File: _docs/reference/os_support.md (hunk 3 above)
Comment (Arau) [reply _docs/reference/os_support.md]: fixed https://github.com/storageos/storageos.github.io/pull/148/commits/93a3652b57840070c18b0153615fe2568a9fcfb2

File: _docs/install/prerequisites/devicepresentation.md (hunk 2 above)
Comment (Arau) [reply _docs/install/prerequisites/devicepresentation.md]: fixed https://github.com/storageos/storageos.github.io/pull/148/commits/93a3652b57840070c18b0153615fe2568a9fcfb2

File: INSTALL_STORE_CONTAINER.md [hunk 5]
```diff
@@ -65,27 +65,12 @@ The container installation method requires Docker 1.10+ running on 64-bit linux.
 
 The node container (or plugin) should be installed on each Docker node where you want to consume StorageOS volumes or to present capacity to other nodes.
 
-### Network Block Device (NBD)
+### Linux-IO (LIO)
 
-(Optional) NBD is a default Linux kernel module that allows block devices to be run in userspace. Enabling NBD is recommended as it will increase performance for some workloads. To enable the module and increase the number of allowable devices, you must either run:
+StorageOS is developed using Linux-IO (LIO) Target. An open-source implementation of the SCSI target.
+However, LIO is supported by most of the kernels available nowadays, some distributions have left the kernel module out of the main kernel package.
```
Comment (oicheryl) [inline INSTALL_STORE_CONTAINER.md]: "However," => "Although"

File: INSTALL_STORE_CONTAINER.md [hunk 6]
```diff
@@ -65,27 +65,12 @@ The container installation method requires Docker 1.10+ running on 64-bit linux.
...
-```
-nbd
-```
-
-2. Add the following module configuration line in `/etc/modprobe.d/nbd.conf`
-
-```
-options nbd nbds_max=1024
-```
+Enabling LIO is mandatory for the best performance, check how to enable LIO here <https://docs.storageos.com/docs/reference/os_support>.
```
Comment (oicheryl) [inline INSTALL_STORE_CONTAINER.md]: Remove "for the best performance".

File: _docs/install/prerequisites/devicepresentation.md [hunk 7]
```diff
@@ -7,51 +7,14 @@ module: install/prerequisites/devicepresentaion.md
 
 # Device presentation
 
-StorageOS requires one of two runtime requirements to present virtual block devices
-from userspace. While not mandatory, the NBD aproach does offer significantly
-better performance over the loop-device alternative.
+StorageOS is developed using Linux-IO (LIO) Target. An open-source implementation of the SCSI target. StorageOS uses a
```
Comment (oicheryl) [inline _docs/install/prerequisites/devicepresentation.md]: . A => , a

File: _docs/install/prerequisites/devicepresentation.md [hunk 8]
```diff
@@ -7,51 +7,14 @@ module: install/prerequisites/devicepresentaion.md
 
 # Device presentation
 
-StorageOS requires one of two runtime requirements to present virtual block devices
-from userspace. While not mandatory, the NBD aproach does offer significantly
-better performance over the loop-device alternative.
+StorageOS is developed using Linux-IO (LIO) Target. An open-source implementation of the SCSI target. StorageOS uses a 
+software interface for Unix-like computer operating systems that lets non-privileged users create their own file systems without editing kernel code.
```
Comment (oicheryl) [inline _docs/install/prerequisites/devicepresentation.md]: Remove the line "StorageOS uses a.... kernel code".

File: _docs/install/prerequisites/devicepresentation.md (hunk 7 above)
Comment (oicheryl) [reply _docs/install/prerequisites/devicepresentation.md]: "is developed using" => "requires the"

File: _docs/install/prerequisites/devicepresentation.md [hunk 9]
```diff
@@ -7,51 +7,14 @@ module: install/prerequisites/devicepresentaion.md
...
-```
-
-To ensure the NBD module is loaded on reboot:
-
-1. Add the following line to `/etc/modules`
-
-    ```text
-    nbd
-    ```
+However, LIO is supported by most of the kernels available nowadays, some distributions have left the kernel module out of the main kernel image. For those, distributions with support,
```
Comment (oicheryl) [inline _docs/install/prerequisites/devicepresentation.md]: "However," => "Although"

Comment (oicheryl) [reply _docs/install/prerequisites/devicepresentation.md]: "For those, distributions with support, will be required to enable kernel modules. "
=>
"For supported distributions, it is mandatory to enable certain kernel module".

File: _docs/install/prerequisites/devicepresentation.md [hunk 10]
```diff
@@ -7,51 +7,14 @@ module: install/prerequisites/devicepresentaion.md
...
+will be required to enable kernel modules. 
 
-1. Add the following module configuration lines in `/etc/modprobe.d/nbd.conf`
+Check the [os distribution support page](/docs/reference/os_support) to validate what OS can be used and how to enable LIO.
```
Comment (oicheryl) [inline _docs/install/prerequisites/devicepresentation.md]: "os" => "OS"
"what" => "which"

File: _docs/install/prerequisites/devicepresentation.md [hunk 11]
```diff
@@ -7,51 +7,14 @@ module: install/prerequisites/devicepresentaion.md
...
-
-While this feature is enabled by default on most systems, StorageOS does require some
-specific parameters to be set. To configure these, add the following
-configuration line in `/etc/modprobe.d/loop.conf`
+## Network Block Device
 
-```text
-options loop max_loop=1024
-```
+StorageOS versions prior 1.0.0 used to use NBD (Network Block Device) to present virtual block devices from userspace. **NBD is no longer used or required by StorageOS.**
```
Comment (oicheryl) [inline _docs/install/prerequisites/devicepresentation.md]: "Network Block Device (deprecated)"
"prior 1.0.0" => "prior to"

File: _docs/reference/os_support.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+layout: guide
+title: StorageOS Docs - OS support
+anchor: reference
+module: reference/ossupport
+---
+
+# OS support
+
+StorageOS uses FUSE known as a Filesystem in Userspace. It is a software interface for Unix-like computer operating systems that lets non-privileged users create their own file systems without editing kernel code. In addition, StorageOS is developed using Linux-IO (LIO) Target. An open-source implementation of the SCSI target.
+However, LIO is supported by most of the kernels available nowadays, some distributions have left the kernel module out of the main kernel package.
```
Comment (oicheryl) [inline _docs/reference/os_support.md]: "However, " => "Although"

File: _docs/reference/os_support.md [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+
+
+## Distribution support
+
+The following table shows which distributions can enable the kernel modules to use LIO, whether needed to install linux-virtual-extra or not.
```
Comment (oicheryl) [inline _docs/reference/os_support.md]: "whether needed to install linux-virtual-extra or not." => "whether extra packages are required".

File: _docs/reference/os_support.md [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+
+chmod +x enable_lio.sh
+sudo ./enable_lio.sh
+```
+
+## Ubuntu generic and GCE
+
+It is required to install linux-image-extra for any generic ubuntu. 
+
+Google Cloud Engine has got a candidate to install linux-image-extra with gce optimisation for Ubuntu 16.04. However, Ubuntu 18.04 doesn't have a package for it yet.
```
Comment (oicheryl) [inline _docs/reference/os_support.md]: "has got" => "has"
". However, Ubuntu 18.04 doesn't have a package for it yet." => "but this package is not yet available for Ubuntu 18.04."

File: _docs/reference/os_support.md [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+
+```
+sudo apt -y update
+sudo apt -y install linux-image-extra-$(uname -r)
+```
+
+## Why there is no support for Ubuntu (AWS, Azure)
+Canonical has created Ubuntu images for cloud providers with optimised versions of the kernel. These versions don't include the kernel modules needed by LIO and the linux-image-extra, where the modules are found in the generic image, doesn't have a candidate for the optimised version of the kernels.
```
Comment (oicheryl) [inline _docs/reference/os_support.md]: "Canonical has created" => "Canonical created"

File: INSTALL_STORE_CONTAINER.md (hunk 5 above)
Comment (Arau) [reply INSTALL_STORE_CONTAINER.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: INSTALL_STORE_CONTAINER.md (hunk 6 above)
Comment (Arau) [reply INSTALL_STORE_CONTAINER.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: _docs/install/prerequisites/devicepresentation.md (hunk 7 above)
Comment (Arau) [reply _docs/install/prerequisites/devicepresentation.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: _docs/install/prerequisites/devicepresentation.md (hunk 8 above)
Comment (Arau) [reply _docs/install/prerequisites/devicepresentation.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: _docs/install/prerequisites/devicepresentation.md (hunk 9 above)
Comment (Arau) [reply _docs/install/prerequisites/devicepresentation.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: _docs/install/prerequisites/devicepresentation.md (hunk 10 above)
Comment (Arau) [reply _docs/install/prerequisites/devicepresentation.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: _docs/install/prerequisites/devicepresentation.md (hunk 11 above)
Comment (Arau) [reply _docs/install/prerequisites/devicepresentation.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: _docs/reference/os_support.md (hunk 12 above)
Comment (Arau) [reply _docs/reference/os_support.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: _docs/reference/os_support.md (hunk 13 above)
Comment (Arau) [reply _docs/reference/os_support.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: _docs/reference/os_support.md (hunk 14 above)
Comment (Arau) [reply _docs/reference/os_support.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

File: _docs/reference/os_support.md (hunk 15 above)
Comment (Arau) [reply _docs/reference/os_support.md]: Thank you! Done https://github.com/storageos/storageos.github.io/pull/148/commits/85f8b90c9002e67efe6eff881dff4b73ed98fad3

**Pull request merged**